File: TsinghuaNet.Eto/TsinghuaNet.Eto/Converters/FluxLimitConverter.vb
Imports System.Globalization
Imports Eto.Forms

Public Class FluxLimitConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        Dim bytes As ByteSize = value
        Return CInt(bytes.GigaBytes)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return ByteSize.FromGigaBytes(value)
    End Function
End Class

File: TsinghuaNet.Eto/TsinghuaNet.Eto/Converters/FluxLimitConverter.cs
using System;
using System.Globalization;
using Eto.Forms;

namespace TsinghuaNet.Eto.Converters
{
    public class FluxLimitConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            ByteSize bytes = (ByteSize)value;
            return bytes.GigaBytes;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return ByteSize.FromGigaBytes((double)value);
        }
    }
}
